certify-service/src/main/java/io/mosip/certify/services/VCIssuanceServiceImpl.java

```diff
@@ -120,14 +123,15 @@ private VCResult<?> getVerifiableCredential(CredentialRequest credentialRequest,
         parsedAccessToken.getClaims().put("accessTokenHash", parsedAccessToken.getAccessTokenHash());
         VCRequestDto vcRequestDto = new VCRequestDto();
         vcRequestDto.setFormat(credentialRequest.getFormat());
-        vcRequestDto.setContext(credentialRequest.getCredential_definition().getContext());
-        vcRequestDto.setType(credentialRequest.getCredential_definition().getType());
-        vcRequestDto.setCredentialSubject(credentialRequest.getCredential_definition().getCredentialSubject());
+
```

Comment:
nitpick(optional): lines 132-134 & 143-146 seem exactly similar can they be moved back outside the switch-case & the try-catch?

This should come as a Sonar error later, hence asked.

Reply:
mso_mdoc format VC does not have credential_definition as part of credential request, for this reason we cannot move it outside switch, as its not a common piece of code for different VC formats, we can try extracting a separate method for this (May be like `populateVCRequestDTOWithCredentialDefinitionDetails`)